addons/mail/static/src/new/utils.js

```diff
@@ -191,3 +199,41 @@ export function useVisible(refName, cb) {
     }
     return state;
 }
+
+export function useSnapshot(refName, { onWillPatch: p_onWillPatch, onPatched: p_onPatched }) {
+    const ref = useRef(refName);
+    const snapshot = {
+        scrollHeight: null,
+        scrollTop: null,
+        clientHeight: null,
+    };
+    onMounted(() => {
+        const el = ref.el;
+        Object.assign(snapshot, {
+            scrollHeight: el.scrollHeight,
+            scrollTop: el.scrollTop,
+            clientHeight: el.clientHeight,
+        });
+    });
+    onWillPatch(() => {
+        const el = ref.el;
+        Object.assign(snapshot, {
+            scrollHeight: el.scrollHeight,
+            scrollTop: el.scrollTop,
+            clientHeight: el.clientHeight,
+        });
+        Object.assign(snapshot, {
+            ...p_onWillPatch(),
+        });
```

Comment:
why not putting it all in the same Object.assign?

Reply:
done